pkg/server/server.go

```diff
@@ -0,0 +1,78 @@
+// package server is an implementation of the remote-exec server.
+package server
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+
+	"github.com/mlaterman/remote-exec/pkg/auth"
+	pb "github.com/mlaterman/remote-exec/pkg/proto"
+)
+
+type ExecServer struct {
+	pb.UnimplementedExecutorServer
+}
+
+func New() *ExecServer {
+	return &ExecServer{}
+}
+
+func (e *ExecServer) checkCert(ctx context.Context) error {
+	cert, err := getClientCert(ctx)
+	if err != nil {
+		return err
+	}
+	name := "" // TODO get name value
```

Comment:
we will extract and track the name with the processes the server starts; none of that is part of this PR